### ISSUE TITLE
Error when checking target: expected flatten_1 to have shape (72,) but got array with shape (2,)

### ISSUE THREAD

**Issue by sid7590** [keras-team/keras#11511]
I have just changed the kernel size and pool size of a working code and I am getting this error " **Error when checking target: expected flatten_1 to have shape (72,) but got array with shape (2,)**" 
my code goes as follows:

model = models.Sequential()
model.add(Conv2D(24, kernel_size=8, strides=2, input_shape=(256, 256, 3)))
model.add(MaxPooling2D(pool_size=(3, 3), strides=2))
model.add(Conv2D(48, kernel_size=6, strides=2))
model.add(MaxPooling2D(pool_size=(3, 3), strides=2))
model.add(Conv2D(96, kernel_size=4, strides=2))
model.add(Activation('relu'))
model.add(Conv2D(2, kernel_size=1, strides=1))
model.add(Activation('softmax'))
model.add(Flatten())

opt = SGD(lr=0.001)

model.compile(loss='categorical_crossentropy', optimizer=opt, metrics=['accuracy'])

model.summary()

checkpoint_callback = ModelCheckpoint(filepath='model_dr_hand.hdf5', monitor='val_acc', save_best_only=True, mode='auto', period=1)

model.fit(training_images, training_labels, epochs=200, batch_size=16, verbose=1,
            validation_data=(validation_images, validation_labels), callbacks=[checkpoint_callback])

#prediction = model.predict(test_images)
rounded_predictions=model.predict_classes(test_images)

for i in range(0, len(rounded_predictions)):
    if rounded_predictions[i] == 0:
        print("This image is a door handle")
    if rounded_predictions[i] == 1:
       print("This image is not a door handle")
